src/pages/Overview/OverviewPage.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { Link } from 'react-router-dom';
+import {
+  AggregateStatusNotifications,
+  AggregateStatusNotification,
+  Breadcrumb,
+  Card,
+  CardBody,
+  CardGrid,
+  CardTitle,
+  Col,
+  Icon,
+  Row
+} from 'patternfly-react';
+import { AxiosError } from 'axios';
+
+import { FilterSelected } from '../../components/Filters/StatefulFilters';
+import { ListPage } from '../../components/ListPage/ListPage';
+import * as API from '../../services/Api';
+import { ActiveFilter } from '../../types/Filters';
+import { AppHealth, FAILURE, DEGRADED, HEALTHY } from '../../types/Health';
+import Namespace from '../../types/Namespace';
+import { authentication } from '../../utils/Authentication';
+
+import { FiltersAndSorts } from './FiltersAndSorts';
+import OverviewStatus from './OverviewStatus';
+import OverviewToolbar from './OverviewToolbar';
+import NamespaceInfo from './NamespaceInfo';
+
+type State = {
+  namespaces: NamespaceInfo[];
+  showEmpty: boolean;
+};
+
+class OverviewPage extends ListPage.Component<{}, State> {
+  static isPassingFilters(info: NamespaceInfo): boolean {
+    const activeFilters: ActiveFilter[] = FilterSelected.getSelected();
+    const nameFilters = activeFilters.filter(f => f.category === FiltersAndSorts.nameFilter.title);
+    const nameOK = nameFilters.length === 0 || nameFilters.some(f => info.name.includes(f.value));
+    if (!nameOK) {
+      return false;
+    }
+    const healthFilters = activeFilters.filter(f => f.category === FiltersAndSorts.healthFilter.title);
+    const statusOK =
+      healthFilters.length === 0 ||
+      healthFilters.some(f => {
+        switch (f.value) {
+          case FAILURE.name:
+            return info.appsInError.length > 0;
+          case DEGRADED.name:
+            return info.appsInWarning.length > 0;
+          case HEALTHY.name:
+            return info.appsInSuccess.length > 0;
+          default:
+            return false;
+        }
+      });
+    return statusOK;
+  }
+
+  constructor(props: RouteComponentProps<{}>) {
+    super(props);
+    this.state = {
+      namespaces: [],
+      showEmpty: false
+    };
+  }
+
+  componentDidMount() {
+    this.load();
+  }
+
+  load = () => {
+    API.getNamespaces(authentication())
+      .then(namespacesResponse => {
+        const namespaces: Namespace[] = namespacesResponse['data'];
+        this.fetchAppsHealth(namespaces.map(namespace => namespace.name), this.currentDuration());
```

Comment:
This is going to filter all namespaces even if filtered.
Or perhaps I missed some part in the code ?

Reply:
Yes, the filter doesn't expect exact matching so I have to get all namespace names before filtering.
However there's indeed some place for improvement, as we could perform the namespace filtering _before_ calling `fetchAppHealth`. I'm going to do that.

Reply:
Done, improved loading sequence.
That makes me have to rewrite tests